### ISSUE TITLE
Rule proposal: `explicit-timer-delay`

### ISSUE THREAD

**Issue by azat-io** [sindresorhus/eslint-plugin-unicorn#2809]
### Description

Enforce or disallow explicit delay argument for `setTimeout()` and `setInterval()`

Allow configuring whether `setTimeout()` and `setInterval()` must have an explicit delay argument. This provides flexibility for different team preferences.

### Examples

```typescript
{
  "unicorn/explicit-timer-delay": ["error", "always" | "never"]
}
```

- `"always"` (default): Require explicit delay argument
- `"never"`: Disallow explicit delay when it's 0 (prefer implicit default)

**With `"always"`:**

```js
// ❌ Fail
setTimeout(() => console.log('Hello'));
setInterval(callback);

// ✅ Pass
setTimeout(() => console.log('Hello'), 0);
setInterval(callback, 0);
```

**With `"never"`:**

```js
// ❌ Fail
setTimeout(() => console.log('Hello'), 0);

// ✅ Pass
setTimeout(() => console.log('Hello'));
setInterval(callback, 1000); // Non-zero is still OK
```

### Proposed rule name

explicit-timer-delay
